### PR TITLE
aws_ec2_transit_gateway_route closes #636

### LINKED ISSUE #636: aws_ec2_transit_gateway_route
Just looking for the ability to  get a list of routes from any/all AWS Transit Gateway route tables

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_ec2_transit_gateway_route []

PRETEST: tests/aws_ec2_transit_gateway_route

TEST: tests/aws_ec2_transit_gateway_route
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ec2_transit_gateway.named_test_resource: Creating...
aws_ec2_transit_gateway.named_test_resource: Still creating... [10s elapsed]
aws_ec2_transit_gateway.named_test_resource: Still creating... [20s elapsed]
aws_ec2_transit_gateway.named_test_resource: Still creating... [30s elapsed]
aws_ec2_transit_gateway.named_test_resource: Still creating... [40s elapsed]
aws_ec2_transit_gateway.named_test_resource: Still creating... [50s elapsed]
aws_ec2_transit_gateway.named_test_resource: Still creating... [1m0s elapsed]
aws_ec2_transit_gateway.named_test_resource: Creation complete after 1m10s [id=tgw-01f7395b7e3bfd332]
aws_ec2_transit_gateway_route.named_test_resource: Creating...
aws_ec2_transit_gateway_route_table.named_test_resource: Creating...
aws_ec2_transit_gateway_route.named_test_resource: Still creating... [10s elapsed]
aws_ec2_transit_gateway_route_table.named_test_resource: Still creating... [10s elapsed]
aws_ec2_transit_gateway_route_table.named_test_resource: Still creating... [20s elapsed]
aws_ec2_transit_gateway_route.named_test_resource: Still creating... [20s elapsed]
aws_ec2_transit_gateway_route.named_test_resource: Still creating... [30s elapsed]
aws_ec2_transit_gateway_route_table.named_test_resource: Still creating... [30s elapsed]
aws_ec2_transit_gateway_route_table.named_test_resource: Still creating... [40s elapsed]
aws_ec2_transit_gateway_route.named_test_resource: Still creating... [40s elapsed]
aws_ec2_transit_gateway_route_table.named_test_resource: Creation complete after 46s [id=tgw-rtb-0b90b2dde7cbb7e85]
aws_ec2_transit_gateway_route.named_test_resource: Still creating... [50s elapsed]
aws_ec2_transit_gateway_route.named_test_resource: Creation complete after 58s [id=tgw-rtb-06d07766168b39ca1_0.0.0.0/0]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = 632902152528
aws_partition = aws
destination_cidr_block = 0.0.0.0/0
region_name = us-east-1
resource_aka = arn:aws:ec2:us-east-1:632902152528:transit-gateway-route-table/tgw-rtb-06d07766168b39ca1:0.0.0.0/0
resource_name = turbottest53255
transit_gateway_rtb_id = tgw-rtb-06d07766168b39ca1

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:ec2:us-east-1:632902152528:transit-gateway-route-table/tgw-rtb-06d07766168b39ca1:0.0.0.0/0"
    ],
    "title": "0.0.0.0/0",
    "transit_gateway_route_table_id": "tgw-rtb-06d07766168b39ca1"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "destination_cidr_block": "0.0.0.0/0",
    "state": "blackhole",
    "transit_gateway_route_table_id": "tgw-rtb-06d07766168b39ca1",
    "type": "static"
  }
]
✔ PASSED

POSTTEST: tests/aws_ec2_transit_gateway_route

TEARDOWN: tests/aws_ec2_transit_gateway_route

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic transit gateway route info

```sql
select
  transit_gateway_route_table_id,
  destination_cidr_block,
  prefix_list_id,
  state,
  type
from
  aws_ec2_transit_gateway_route;
```

```
+--------------------------------+------------------------+----------------+-----------+--------+
| transit_gateway_route_table_id | destination_cidr_block | prefix_list_id | state     | type   |
+--------------------------------+------------------------+----------------+-----------+--------+
| tgw-rtb-0fbd0797f37e06c07      | 10.0.30.0/28           | <null>         | active    | static |
| tgw-rtb-0fbd0797f37e06c07      | 10.0.0.0/24            | <null>         | blackhole | static |
+--------------------------------+------------------------+----------------+-----------+--------+
```

### List active routes

```sql
select
  transit_gateway_route_table_id,
  destination_cidr_block,
  state,
  type
from
  aws_ec2_transit_gateway_route
where
  state = 'active';
```

```
+--------------------------------+------------------------+--------+--------+
| transit_gateway_route_table_id | destination_cidr_block | state  | type   |
+--------------------------------+------------------------+--------+--------+
| tgw-rtb-0fbd0797f37e06c07      | 10.0.30.0/28           | active | static |
+--------------------------------+------------------------+--------+--------+
```

</details>

### CODE REVIEW

File: aws-test/tests/aws_ec2_transit_gateway_route/variables.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "nagraj"
```
Comment (khushboo9024) [inline aws-test/tests/aws_ec2_transit_gateway_route/variables.tf:10]:
```suggestion
  default     = "default"
```

File: aws/table_aws_ec2_transit_gateway_route.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsEc2TransitGatewayRoute(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ec2_transit_gateway_route",
+		Description: "AWS EC2 Transit Gateway Route Table",
```
Comment (khushboo9024) [inline aws/table_aws_ec2_transit_gateway_route.go:18]:
```suggestion
		Description: "AWS EC2 Transit Gateway Route",
```

File: aws/table_aws_ec2_transit_gateway_route.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsEc2TransitGatewayRouteAka,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+type RouteDetails struct {
+	Route                      *ec2.TransitGatewayRoute
```
Comment (khushboo9024) [inline aws/table_aws_ec2_transit_gateway_route.go:78]: Please fix spacing.

File: aws/table_aws_ec2_transit_gateway_route.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "transit_gateway_route_table_id",
+				Description: "The ID of the transit gateway route table.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "destination_cidr_block",
+				Description: "The CIDR block used for destination matches.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_ec2_transit_gateway_route.go:33]: seems like the property stores `CIDR` range. If so, use `proto.ColumnType_CIDR` instead.

**Review by khushboo9024 — APPROVED** [review]

**Pull request merged**